input/input_test.go

```diff
@@ -14,6 +15,171 @@ import (
 	"github.com/raintank/schema"
 )
 
+func TestIngestValidTagAndValueWithRejection(t *testing.T) {
+	handler, index, reset := getDefaultHandler(t)
+	defer reset()
+
+	rejectInvalidTags = false
+	data := getTestMetricData()
+	data.Tags = []string{"valid=tag"}
+	testIngestMetricData(t, "valid_with_rejection_0", data, handler, index, 0, 0, 1)
```

Comment:
why don't we follow the best practice of 1 function that has all the cases listed as entries in a table?

Reply:
Just because it results in clearer errors when one fails. but I can easily change that

Reply:
also, this is a lot of code for testing something that should be fairly trivial. especially all the code around dynamically generating test data, do we need it?

Reply:
It definitely helps to make sure that the logic does what we want it to, and to ensure it doesn't get changed by accident in the future for example due to changes in raintank/schema.